### ISSUE TITLE
sphinx-quickstart asks again and again Y|N on MS DOS / MS Windows

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1043]
When run on MS DOS / MS Windows (Command line or MinGW32), sphinx-quickstart loops on boolean questions:

```
#!shell

You have two options for placing the build directory for Sphinx output.
Either, you use a directory "_build" within the root path, or you separate
"source" and "build" directories within the root path.
> Separate source and build directories (y/N) [n]:
* Please enter either 'y' or 'n'.
> Separate source and build directories (y/N) [n]: n
* Please enter either 'y' or 'n'.
> Separate source and build directories (y/N) [n]: N
* Please enter either 'y' or 'n'.
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1043
- Originally reported by: Régis Décamps
- Originally created at: 2012-11-26T11:02:57.386

**Comment by shimizukawa** [comment]
_From Régis Décamps on 2012-11-26 10:30:21+00:00_

The reason is that the `input()` methods returns the return character `\r`.
I sugguest this easy fix:

```
def do_prompt(d, key, text, default=None, validator=nonempty):
    while True:
        if default:
            prompt = purple(PROMPT_PREFIX + '%s [%s]: ' % (text, default))
        else:
            prompt = purple(PROMPT_PREFIX + text + ': ')
-        x = term_input(prompt)
+        x = term_input(prompt).strip()
        if default and not x:
            x = default
```

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2012-11-27 23:04:39+00:00_

I could not reproduce this behavior in  my Windows7(cmd and msys bash).
But [Régis Décamps](https://bitbucket.org/regis)'s patch seems reasonable enough and no-side-effect.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2012-11-28 00:54:45+00:00_

I remembered that it was the issue of Python3.2 on Windows.

-- http://bugs.python.org/issue11272

```
 You are using the only version that has the bug (3.2.0, also called 3.2).
 The fixed version will be released shortly (3.2.1).
```

**Comment by shimizukawa** [comment]
_From Jon Waltman on 2012-11-28 02:34:58+00:00_

I think [Régis Décamps](https://bitbucket.org/regis)'s patch looks fine as well.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2012-11-28 07:08:51+00:00_

I'll check on my Windows + Python3.2.0 ASAP.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2012-11-28 14:22:42+00:00_

fix: sphinx-quickstart asks again and again Y|N because input() return value with extra '\r' on Python-3.2.0 + Windows. closes #1043

→ <<cset e9c058271319baa0e586b4bca771768168fc7343>>
